src/runner/sqs.ts

```diff
@@ -147,24 +159,26 @@ class SqsRunner extends BaseRunner implements IRunner {
       `Polling for messages (${topics ? topics.join(',') : 'all'})`
     );
 
-    await Promise.all(
-      subscriptions.map(async topic => {
+    const config = this.config as SQSConfiguration;
+    await bluebird.map(
+      subscriptions,
+      async topic => {
         const queueURL = await this.getQueueUrl(topic);
         if (queueURL) {
           this.logger.debug(`starting processing of ${topic} with ${queueURL}`);
+
           const params = {
-            MaxNumberOfMessages: (this.config as SQSConfiguration)
-              .maxNumberOfMessages,
+            MaxNumberOfMessages: config.maxNumberOfMessages,
             QueueUrl: queueURL,
-            VisibilityTimeout: (this.config as SQSConfiguration)
-              .visibilityTimeout,
-            WaitTimeSeconds: (this.config as SQSConfiguration).waitTimeSeconds,
+            VisibilityTimeout: config.visibilityTimeout,
+            WaitTimeSeconds: config.waitTimeSeconds,
           };
           await this.dequeue(topic, params);
         } else {
           this.logger.error(`Queue URL ${topic} not found`);
         }
-      })
+      },
+      { concurrency: this.concurrency }
```

Comment:
I had been going to default this to subscriptions.length for 1 at a time message per subscription processing by default. Not sure if it'd add a huge amount at this point though

Reply:
@rrsmith yeah I want to make it smarter, potentially depending on WorkerConfig for the concurrency options.

Let me revisit in the next PR.

Reply:
Sounds good 👍